skgstat/interfaces/gstools.py

```diff
@@ -1,46 +1,56 @@
-try:
-    import gstools
-    GSTOOLS_AVAILABLE = True
-except ImportError:
-    GSTOOLS_AVAILABLE = False
+"""GSTools Interface."""
 import numpy as np
 
 
-def __check_gstools_available():  # pragma: no cover
-    if not GSTOOLS_AVAILABLE:
-        print('The gstools interface needs gstools installed.')
-        print("Run 'pip install gstools' to install it.")
-        return False
-    return True
-
-
-def gstools_cov_model(variogram, **kwargs):
-    """GSTools Interface
-
-    Pass a :class:`skgstat.Variogram` instance.
-    Returns an **already fitted** variogram model
-    inherited from `gstools.CovModel`. All
-    kwargs passed will be passed to
-    `gstools.CovModel`.
-
-    """
-    # extract the fitted variogram model
-    fitted_model = variogram.fitted_model
-
-    # define the CovModel
-    class VariogramModel(gstools.CovModel):
-        def variogram(self, r):
-            if isinstance(r, np.ndarray):
-                return fitted_model(r.flatten()).reshape(r.shape)
-            else:
-                return fitted_model(r)
-
-    # dim can be infered from variogram
-    if 'dim' not in kwargs.keys():
-        kwargs['dim'] = variogram.coordinates.ndim
-
-    # Create the instance
-    model = VariogramModel(**kwargs)
-    model.fit_variogram(variogram.bins, variogram.experimental)
-
-    return model
+def stable_rescale(describe):
+    """Get GSTools rescale parameter from sk-gstat stable model description."""
+    return np.power(3, 1 / describe["shape"])
+
+
+def matern_rescale(describe):
+    """Get GSTools rescale parameter from sk-gstat matern model description."""
+    if 0.5 < describe["smoothness"] < 10.0:
+        return 4.
+    return 6.
+
+
+MODEL_MAP = dict(
+    spherical=dict(gs_cls="Spherical"),
+    exponential=dict(gs_cls="Exponential", rescale=3.0),
+    gaussian=dict(gs_cls="Gaussian", rescale=2.0),
+    cubic=dict(gs_cls="Cubic"),
+    stable=dict(
+        gs_cls="Stable", arg_map={"alpha": "shape"}, rescale=stable_rescale
+    ),
+    matern=dict(
+        gs_cls="Matern", arg_map={"nu": "smoothness"}, rescale=matern_rescale
+    ),
+)
+
+
+def skgstat_to_gstools(variogram, **kwargs):
+    """Instantiate a corresponding GSTools CovModel."""
+    try:
+        import gstools as gs
+    except ImportError:
+        raise ImportError("skgstat_to_gstools: GSTools needs to be installed.")
+    kwargs.setdefault("dim", variogram.coordinates.ndim)
```

Comment:
Is that really the spatial dimension? Just copied this line from the old interface.

Reply:
Probably not. `Variogram.coordinates` points to a numpy.ndarray of shape `(n_obs, n_dims)`

I.e. for 100 3D coords the shape will be `(300, 3)`. Thus the `ndim` would be `2` (I think) and `variogram.coordinates.shape[1]` is the actual spatial dimension, right?

Thanks for pointing out.

Reply:
You mean `(100, 3)` for 100 3D values, right? OK, for 1D I need to assure that there is an array with 2 axis.
```python
np.array(variogram.coordinates, ndmin=2).shape[1]
```
Or you could already convert this on creation for the Variogram class. Maybe you could also create an attribute holding the dimension doing this internally, so we just use `variogram.dim` here.

Reply:
Yeah `(100,3)`.
I'll go for the `dim`. If I am not mistaken, 1D arrays are transformed to (x, 2) anyway, but `Variogram` takes the property approach everywhere anyway, so having a `Variogram.dim` makes much sense and looks quite clean.

Reply:
AFAICT, it is not transformed inplace, so `variogram._X` will be a 1D array if given as 1D array. So you could go with:
```python
@property
def dim(self):
    return np.array(self._X, ndmin=2).shape[1]
```
I don't know what happens for `coordinates=None`.

Reply:
Just realized the same thing. Right now, @redhog is writing a class to wraps input coordinates internally, that handles sampling for a very large input dataset.
Thus, the current implementation of `_X` will change soon. Therefore, using properties like `coordinates` and `dim` is the best way to produce the same output when the implementation under the hood changes.

For `Variogram` passing coordinates is mandatory. Thus, `coordinates=None` is kind of wrong usage, but I think I will add an `AttributeError` if coordinates is none.

Reply:
Found it: 

https://github.com/mmaelicke/scikit-gstat/blob/a2bf354a4b3243453c11a6b3154709390858f89c/skgstat/Variogram.py#L1395

Oh boy, not a super good style to hide that on line 1395....
However, I'll go for a `dim` property that handles the dimensionality correct and make sure that it output doesn't change no matter what we implement under the hood...

Reply:
`Variogram.dim` is now implemented with version `0.4.3`.

Reply:
> Found it:
> 
> https://github.com/mmaelicke/scikit-gstat/blob/a2bf354a4b3243453c11a6b3154709390858f89c/skgstat/Variogram.py#L1395
> 
> Oh boy, not a super good style to hide that on line 1395....
> However, I'll go for a `dim` property that handles the dimensionality correct and make sure that it's output doesn't change no matter what we implement under the hood...

Saw that as well. But `self._X` is not overwritten, so this shouldn't harm the proposed `dim` attribute.